annotation-utils/src/main/java/edu/isi/vista/annotationutils/ExtractAnnotationStats.kt

```diff
@@ -0,0 +1,223 @@
+ * the updated annotation files.
+ */
+class ExtractAnnotationStats {
+    companion object {
+        fun main(argv: Array<String>) {
+            if (argv.size != 1) {
+                throw RuntimeException("Expected a single argument: a parameter file")
+            }
+            val paramsLoader = SerifStyleParameterFileLoader.Builder()
+                    .interpolateEnvironmentalVariables(true).build()
+            val params = paramsLoader.load(File(argv[0]))
+            extractStats(params)
+        }
+        fun extractStats(params: edu.isi.nlp.parameters.Parameters) {
+            val exportAnnotationRoot = params.getExistingDirectory("exportedAnnotationRoot")
+            val statisticsDirectory = params.getExistingDirectory("statisticsDirectory")
+            val currentDate = Calendar.getInstance(TimeZone.getTimeZone("UTC"))
+            TimeZone.setDefault(TimeZone.getTimeZone("UTC"))
+            val formatter = SimpleDateFormat("yyyy-MM-dd")
+            val thisDate = formatter.format(currentDate.getTime())
+
+            val annotationSheet: MutableList<Annotation> = arrayListOf()
+
+            // Go through each file in each user's folder and collect the stats
+            var user: String? = null
+            var subtype: String? = null
+            File(exportAnnotationRoot.toString()).walk().forEach {
+                // Skip irrelevant dirs
+                if (it == File(exportAnnotationRoot.toString()) ||
+                        it.name.contains("""sandbox""") ||
+                        it.name.contains("""gigaword""")) {
+                }
+                else if (it.isDirectory) {
+                    val folder = it.name
+                    // Collect username, subtype, and subSubtype
+                    user = Regex(pattern = """\.\w*-(\w*)""")
+                            .find(input = folder.toString())!!.groupValues[1]
+                    subtype = Regex(pattern = """(\w*\.\w*)-""")
+                            .find(input = folder.toString())!!.groupValues[1]
+                }
+                else if (it.isFile) {
+                    if (user.isNullOrBlank() || subtype.isNullOrBlank()) {
+                        throw RuntimeException("Missing annotation information (user/subtype)")
+                    }
+                    val documentName: String = it.toString()
+                    // TODO: count events rather than annotated documents
+                    // This is sort of a placeholder for a better method
+                    val jsonString: String = File(it.toString()).readText(Charsets.UTF_8)
+                    if (jsonString.contains(Regex(""""CTEventSpan" : \["""))) {
+                        annotationSheet.add(Annotation(documentName, user!!, subtype!!))
+                    }
+                }
+            }
+            // From annotationSheet, get TOTAL, TOTAL PER USER/SUBTYPE/SUBSUBTYPE
+            val totalAnnotations = annotationSheet.size
```

Comment:
I am confused by the `Sheet` in the name `annotationSheet`. Perhaps clearer would be `documentsWithEventAnnotations`?